crates/bevy_core/src/name.rs

```diff
@@ -76,6 +78,25 @@ impl std::fmt::Display for Name {
     }
 }
 
+/// Convenient query for giving a human friendly name to an entity.
+#[derive(WorldQuery)]
+pub struct DebugName {
+    /// A [`Name`] that the entity might have that is displayed if available.
+    pub name: Option<&'static Name>,
+    /// The unique identifier of the entity as a fallback.
+    pub entity: Entity,
+}
+
+impl std::fmt::Debug for DebugName {
+    #[inline(always)]
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self.name {
+            Some(name) => std::fmt::Debug::fmt(&name, f),
```

Comment:
maybe it would be worth printing out the `self.entity` too since `Name` isnt required to be a unique identifier